WordPress/gutenberg
# [Button Block]: Add padding block support

## Description

Adds padding support to the Button block using the spacing block support.
Issue #31731

**Notes:**
The block support currently allows arbitrarily large padding values which can cause blocks implementing it to overflow the page. See #31773

## How has this been tested?



Manually.

**Test Setup**
Your theme will need to opt into padding support. You can do this by updating your `theme.json` to enable padding under the spacing control, in either your default settings or the block context for button:
```
"settings": {
    "spacing": {
        "customPadding": true,
        "units": [
            "px",
            "em",
            "rem",
            "vh",
            "vw"
        ]
    }
}
```

**Test Instructions**

1. Create a post, add a buttons block with a button.
2. Select the button block and confirm padding controls in the sidebar work, adjusting units/padding per side.
3. Add another button and confirm that its padding is default. Update its padding and verify that it does not affect the padding of the first button.
3. Save the post and confirm the padding is visually correct on the frontend.

## Screenshots 
![Screen Capture on 2021-05-27 at 13-55-29](https://user-images.githubusercontent.com/63313398/119896035-522ddd80-bef3-11eb-889e-417861d7ea9a.gif)

## Types of changes





## Checklist:
